core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java

```diff
@@ -71,7 +72,7 @@ public ColumnPageDecoder createDecoder(List<Encoding> encodings, List<ByteBuffer
     byte[] encoderMeta = encoderMetas.get(0).array();
     ByteArrayInputStream stream = new ByteArrayInputStream(encoderMeta);
     DataInputStream in = new DataInputStream(stream);
-    if (encoding == DIRECT_COMPRESS) {
+    if (encoding == DIRECT_COMPRESS || encoding == DIRECT_COMPRESS_VARCHAR) {
```

Comment:
why DIRECT_COMPRESS_VARCHAR is required?? you can use DIRECT_COMPRESS for varchar

Reply:
`DIRECT_COMPRESS_VARCHAR` is a special encoding to distinguish normal string and long string. It is used in CompressedDimensionChunkFileBasedReaderV3 to indicate the DimensionStoreType.

Reply:
@xuchuanyin I think in your last PR it was VARCHAR...